src/main/java/org/shanerx/tradeshop/Utils.java

```diff
@@ -46,219 +55,381 @@
+		int count = 0;
+		for (ItemStack itm : inv.getContents()) {
+			if (itm != null) {
+				if (itm.getType() == mat && itm.getDurability() == durability) {
+					count += itm.getAmount();
+				}
+			}
+		}
+		return count >= amt;
+	}
+	
+	public String colorize(String msg) {
+		msg = ChatColor.translateAlternateColorCodes('&', msg);
+		return msg;
+	}
+	
+	public Sign findShopSign(Block chest) {
+		ArrayList<BlockFace> faces = plugin.getAllowedDirections();
+		Collections.reverse(faces);
+		
+		for (BlockFace face : faces) {
+			Block relative = chest.getRelative(face);
+			if (isSign(relative))
+				if (isShopSign(relative))
+					return (Sign) chest.getRelative(face).getState();
+		}
+		
+		
+		return null;
+		
+	}
+	
+	public Block findShopChest(Block sign) {
+		ArrayList<Material> invs = plugin.getAllowedInventories();
+		ArrayList<BlockFace> faces = plugin.getAllowedDirections();
+		
+		for (BlockFace face : faces) {
+			Block relative = sign.getRelative(face);
+			if (relative != null)
+				if (invs.contains(relative.getType()))
+					return sign.getRelative(face);
+		}
+		return null;
+	}
+	
+	public List<OfflinePlayer> getShopOwners(Block b) {
+		TradeShop ts = (TradeShop) Bukkit.getPluginManager().getPlugin("TradeShop");
+		if (!ts.getAllowedInventories().contains(b.getType())) {
+			return null;
+		}
+		
+		List<OfflinePlayer> owners = new ArrayList<>();
+		Inventory inv = ((InventoryHolder) b.getState()).getInventory();
+		String names = inv.getName();
+		for (String m : names.split(";")) {
+			if (m.startsWith("o:")) {
+				owners.add(Bukkit.getOfflinePlayer(m.substring(2)));
+			}
+		}
+		Sign s = findShopSign(b);
+		if (s.getLine(3) == null || s.getLine(3).equals("")) {
+			return owners;
+		}
+		else if (!owners.contains(Bukkit.getOfflinePlayer(s.getLine(3)))) {
+			owners.add(Bukkit.getOfflinePlayer(s.getLine(3)));
+		}
+		return owners;
+	}
+	
+	public List<OfflinePlayer> getShopMembers(Block b) {
+		TradeShop ts = (TradeShop) Bukkit.getPluginManager().getPlugin("TradeShop");
+		if (!ts.getAllowedInventories().contains(b.getType())) {
+			return null;
+		}
+		
+		List<OfflinePlayer> members = new ArrayList<>();
+		Inventory inv = ((InventoryHolder) b.getState()).getInventory();
+		String names = inv.getName();
+		for (String m : names.split(";")) {
+			members.add(Bukkit.getOfflinePlayer(m.substring(2)));
+		}
+		Sign s = findShopSign(b);
+		if (s.getLine(3) == null || s.getLine(3).equals("")) {
+			return members;
+		}
+		else if (!members.contains(Bukkit.getOfflinePlayer(s.getLine(3)))) {
+			members.add(Bukkit.getOfflinePlayer(s.getLine(3)));
+		}
+		return members;
+	}
+	
+	public List<OfflinePlayer> getShopOwners(Sign s) {
+		Chest c = (Chest) findShopChest(s.getBlock());
+		if (c == null) {
+			return null;
+		}
+		return getShopOwners(c.getBlock());
+	}
+	
+	public List<OfflinePlayer> getShopMembers(Sign s) {
+		Chest c = (Chest) findShopChest(s.getBlock());
```

Comment:
Fixed in [this commit](https://github.com/SparklingComet/TradeShop/pull/19/commits/b4fd175bbadc031d7420a5ae2eeb638209826fc9)